### ISSUE TITLE
Integration fails to initialize

### ISSUE THREAD

**Issue by timo-mart** [vmakeev/huawei_mesh_router#10]
I've followed issue #1 as well as I could and I still can't get the integration to start. As far as I can tell, it seems like a separate bug. For me the authentication seems to work, unlike in #1, and the integration appears in the menu, but keeps showing this error:
![image](https://user-images.githubusercontent.com/46269205/201484304-37df5ac9-3e7f-40b3-8817-ea8b5af73f5f.png)
![image](https://user-images.githubusercontent.com/46269205/201484747-f3a8714e-d281-4ec6-976a-6a4d078c393a.png)
```

Unexpected error fetching Huawei AX3 data: Expecting value: line 1 column 1 (char 0)

Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 205, in _async_refresh
    self.data = await self._async_update_data()
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 164, in _async_update_data
    return await self.update_method()
  File "/config/custom_components/huawei_mesh_router/update_coordinator.py", line 125, in async_update
    await self._update_nfc_switch()
  File "/config/custom_components/huawei_mesh_router/update_coordinator.py", line 149, in _update_nfc_switch
    data = await self._api.get("api/bsp/nfc_switch")
  File "/config/custom_components/huawei_mesh_router/huaweiapi.py", line 303, in get
    result = await _get_response_json(response)
  File "/config/custom_components/huawei_mesh_router/huaweiapi.py", line 81, in _get_response_json
    result = json.loads(text)
  File "/usr/local/lib/python3.10/json/__init__.py", line 346, in loads
    return _default_decoder.decode(s)
  File "/usr/local/lib/python3.10/json/decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/usr/local/lib/python3.10/json/decoder.py", line 355, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
```
Thanks in advance!

**Comment by timo-mart** [comment]
Here are also the full logs: 
```
2022-11-12 18:38:34.105 WARNING (SyncWorker_0) [homeassistant.loader] We found a custom integration huawei_mesh_router which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant
2022-11-12 18:40:44.135 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] New instance of HuaweiApi created
2022-11-12 18:40:44.135 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] Authentication started
2022-11-12 18:40:44.135 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] Refresh session called
2022-11-12 18:40:44.136 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] Session created
2022-11-12 18:40:44.136 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] Getting index
2022-11-12 18:40:44.136 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] Performing GET to html/index.html#/login
2022-11-12 18:40:44.341 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] GET html/index.html#/login performed, status: 200
2022-11-12 18:40:44.348 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] Csrf updated: param is 'W4N9wf9VxjFToxggz69L360T0RO0hF9y', token is 'kCUEjF5oHisLwzHbarF4jCfOu00kXwAb'
2022-11-12 18:40:44.349 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] Session cookies stored
2022-11-12 18:40:44.350 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] Sending nonce
2022-11-12 18:40:44.350 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] Performing POST to api/system/user_login_nonce
2022-11-12 18:40:44.362 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] POST to api/system/user_login_nonce performed, status: 200
2022-11-12 18:40:44.362 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] Csrf updated: param is 'Kc9er8zVVH9MkeKO8YouQDy8EWxTcE3S', token is '0EIJDmCsahyPLC0ed0A4df0jxos4VvKD'
2022-11-12 18:40:44.367 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] Sending proof
2022-11-12 18:40:44.368 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] Performing POST to api/system/user_login_proof
2022-11-12 18:40:44.382 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] POST to api/system/user_login_proof performed, status: 200
2022-11-12 18:40:44.382 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] Csrf updated: param is 'rgTns29d70sEy9EPtJP0X0No10n000sN', token is 'uN757i7RuCAF4ZQYtIq3Gr5zBahYhC4h'
2022-11-12 18:40:44.382 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] Authentication success
2022-11-12 18:40:44.383 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] Disconnecting
2022-11-12 18:40:44.397 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] New instance of HuaweiApi created
2022-11-12 18:40:44.397 DEBUG (MainThread) [custom_components.huawei_mesh_router.update_coordinator] Update started
2022-11-12 18:40:44.398 DEBUG (MainThread) [custom_components.huawei_mesh_router.update_coordinator] Updating router data
2022-11-12 18:40:44.398 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] Authentication started
2022-11-12 18:40:44.398 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] Refresh session called
2022-11-12 18:40:44.399 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] Session created
2022-11-12 18:40:44.399 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] Getting index
2022-11-12 18:40:44.399 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] Performing GET to html/index.html#/login
2022-11-12 18:40:44.477 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] GET html/index.html#/login performed, status: 200
2022-11-12 18:40:44.481 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] Csrf updated: param is 'P0Jz7ceVQwM7trN3lYIEvHPP8gwPtjKQ', token is 'Fj6iOvdtbpu7RsrmfWEGOeIyHfMo2Pyw'
2022-11-12 18:40:44.482 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] Session cookies stored
2022-11-12 18:40:44.483 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] Sending nonce
2022-11-12 18:40:44.483 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] Performing POST to api/system/user_login_nonce
2022-11-12 18:40:44.493 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] POST to api/system/user_login_nonce performed, status: 200
2022-11-12 18:40:44.493 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] Csrf updated: param is '0XEZqF2cbeDejkkizLktbp71Lpxy64hs', token is 'HuhgLEsrEZ1EE0OEMFRaqlw76V7vqUjk'
2022-11-12 18:40:44.498 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] Sending proof
2022-11-12 18:40:44.498 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] Performing POST to api/system/user_login_proof
2022-11-12 18:40:44.513 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] POST to api/system/user_login_proof performed, status: 200
2022-11-12 18:40:44.513 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] Csrf updated: param is '1FNIugKXB3LrPdkSmLmYbk7Q5TFzlaFr', token is 'fFwyDSwbnnNd11pU95H5OPMoZ5T2nYqq'
2022-11-12 18:40:44.514 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] Authentication success
2022-11-12 18:40:44.514 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] Performing GET to api/system/deviceinfo
2022-11-12 18:40:44.531 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] GET api/system/deviceinfo performed, status: 200
2022-11-12 18:40:44.534 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] Csrf updated: param is 'feJ30z00tA5XPOO4y5jBGSndGVbc43Zw', token is 'JdG0JnuHhT46fs1Mgsb5tkX3PjsRtzZg'
2022-11-12 18:40:44.535 DEBUG (MainThread) [custom_components.huawei_mesh_router.update_coordinator] Router data updated
2022-11-12 18:40:44.535 DEBUG (MainThread) [custom_components.huawei_mesh_router.update_coordinator] Updating nfc switch state
2022-11-12 18:40:44.535 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] Performing GET to api/bsp/nfc_switch
2022-11-12 18:40:44.543 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] GET api/bsp/nfc_switch performed, status: 404
2022-11-12 18:40:44.543 ERROR (MainThread) [custom_components.huawei_mesh_router.update_coordinator] Unexpected error fetching Huawei AX3 data: Expecting value: line 1 column 1 (char 0)
File "/config/custom_components/huawei_mesh_router/update_coordinator.py", line 125, in async_update
File "/config/custom_components/huawei_mesh_router/update_coordinator.py", line 149, in _update_nfc_switch
File "/config/custom_components/huawei_mesh_router/huaweiapi.py", line 303, in get
File "/config/custom_components/huawei_mesh_router/huaweiapi.py", line 81, in _get_response_json
2022-11-12 18:40:44.560 DEBUG (MainThread) [custom_components.huawei_mesh_router.update_coordinator] Finished fetching Huawei AX3 data in 0.162 seconds (success: False)
2022-11-12 18:40:44.561 WARNING (MainThread) [homeassistant.config_entries] Config entry 'Huawei AX3' for huawei_mesh_router integration not ready yet: Expecting value: line 1 column 1 (char 0); Retrying in background
2022-11-12 18:40:49.568 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] New instance of HuaweiApi created
2022-11-12 18:40:49.569 DEBUG (MainThread) [custom_components.huawei_mesh_router.update_coordinator] Update started
2022-11-12 18:40:49.569 DEBUG (MainThread) [custom_components.huawei_mesh_router.update_coordinator] Updating router data
2022-11-12 18:40:49.570 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] Authentication started
2022-11-12 18:40:49.570 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] Refresh session called
2022-11-12 18:40:49.571 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] Session created
2022-11-12 18:40:49.572 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] Getting index
2022-11-12 18:40:49.572 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] Performing GET to html/index.html#/login
2022-11-12 18:40:49.653 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] GET html/index.html#/login performed, status: 200
2022-11-12 18:40:49.657 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] Csrf updated: param is 'EXEvuFSEEKayX8XO1KAGIhohIUuGyyPz', token is '3zM05nfxcGavJqWeyWkwa5CuqQ0y5Uju'
2022-11-12 18:40:49.658 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] Session cookies stored
2022-11-12 18:40:49.658 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] Sending nonce
2022-11-12 18:40:49.659 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] Performing POST to api/system/user_login_nonce
2022-11-12 18:40:49.671 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] POST to api/system/user_login_nonce performed, status: 200
2022-11-12 18:40:49.671 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] Csrf updated: param is 'S3DLkCtj0iDgzmMmhY5V31aWBg0Xxabm', token is 'nFHZkLg5bwIV75qS5RG6xw1y76nBCyZ1'
2022-11-12 18:40:49.676 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] Sending proof
2022-11-12 18:40:49.677 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] Performing POST to api/system/user_login_proof
2022-11-12 18:40:49.689 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] POST to api/system/user_login_proof performed, status: 200
2022-11-12 18:40:49.690 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] Csrf updated: param is 'JQRVX9SnTStI2g2V742ffkbxoU8LQ8m9', token is 'aweOdXBWD5FgpOrKRieQFguhRpl0OyoL'
2022-11-12 18:40:49.690 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] Authentication success
2022-11-12 18:40:49.690 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] Performing GET to api/system/deviceinfo
2022-11-12 18:40:49.707 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] GET api/system/deviceinfo performed, status: 200
2022-11-12 18:40:49.711 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] Csrf updated: param is 'jjNhLFuk8Eoe2hOXU1Wd0mFlUlEDnE8h', token is 'kw0Z0B3D4Q1GiHa0tvh32pnZGToxxJbE'
2022-11-12 18:40:49.712 DEBUG (MainThread) [custom_components.huawei_mesh_router.update_coordinator] Router data updated
2022-11-12 18:40:49.712 DEBUG (MainThread) [custom_components.huawei_mesh_router.update_coordinator] Updating nfc switch state
2022-11-12 18:40:49.712 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] Performing GET to api/bsp/nfc_switch
2022-11-12 18:40:49.721 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] GET api/bsp/nfc_switch performed, status: 404
2022-11-12 18:40:49.721 ERROR (MainThread) [custom_components.huawei_mesh_router.update_coordinator] Unexpected error fetching Huawei AX3 data: Expecting value: line 1 column 1 (char 0)
File "/config/custom_components/huawei_mesh_router/update_coordinator.py", line 125, in async_update
File "/config/custom_components/huawei_mesh_router/update_coordinator.py", line 149, in _update_nfc_switch
File "/config/custom_components/huawei_mesh_router/huaweiapi.py", line 303, in get
File "/config/custom_components/huawei_mesh_router/huaweiapi.py", line 81, in _get_response_json
2022-11-12 18:40:49.730 DEBUG (MainThread) [custom_components.huawei_mesh_router.update_coordinator] Finished fetching Huawei AX3 data in 0.161 seconds (success: False)
2022-11-12 18:40:59.732 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] New instance of HuaweiApi created
2022-11-12 18:40:59.733 DEBUG (MainThread) [custom_components.huawei_mesh_router.update_coordinator] Update started
2022-11-12 18:40:59.733 DEBUG (MainThread) [custom_components.huawei_mesh_router.update_coordinator] Updating router data
2022-11-12 18:40:59.733 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] Authentication started
2022-11-12 18:40:59.733 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] Refresh session called
2022-11-12 18:40:59.734 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] Session created
2022-11-12 18:40:59.734 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] Getting index
2022-11-12 18:40:59.734 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] Performing GET to html/index.html#/login
2022-11-12 18:40:59.803 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] GET html/index.html#/login performed, status: 200
2022-11-12 18:40:59.806 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] Csrf updated: param is 'ALM6SOsoV1uDFFUsJJd0DhIgmgHaUrCm', token is 'pYoQV0B4Q1slHQc657yal16pPhjAISkf'
2022-11-12 18:40:59.807 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] Session cookies stored
2022-11-12 18:40:59.807 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] Sending nonce
2022-11-12 18:40:59.807 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] Performing POST to api/system/user_login_nonce
2022-11-12 18:40:59.814 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] POST to api/system/user_login_nonce performed, status: 200
2022-11-12 18:40:59.815 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] Csrf updated: param is 'u5NiQtXTo24ISdzddfckWHWflSv3OCFr', token is 'RdFQx0V3jbjChbSxAgZO89iIqCokBCLg'
2022-11-12 18:40:59.817 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] Sending proof
2022-11-12 18:40:59.817 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] Performing POST to api/system/user_login_proof
2022-11-12 18:40:59.826 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] POST to api/system/user_login_proof performed, status: 200
2022-11-12 18:40:59.826 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] Csrf updated: param is 'vSTvV0NcxZPZm4QEkFYOc7iztsGEP3Lk', token is 'Tiso2iZIlIJ2QOBaGN9BJSi50OP4bQyW'
2022-11-12 18:40:59.827 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] Authentication success
2022-11-12 18:40:59.827 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] Performing GET to api/system/deviceinfo
2022-11-12 18:40:59.841 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] GET api/system/deviceinfo performed, status: 200
2022-11-12 18:40:59.843 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] Csrf updated: param is 'jy1Q6rQzfcs5y50hlELqMQApC3lS8Rre', token is 'v1ds0gYuz4C4Ebf54szkyN3HdCLMY90N'
2022-11-12 18:40:59.844 DEBUG (MainThread) [custom_components.huawei_mesh_router.update_coordinator] Router data updated
2022-11-12 18:40:59.844 DEBUG (MainThread) [custom_components.huawei_mesh_router.update_coordinator] Updating nfc switch state
2022-11-12 18:40:59.844 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] Performing GET to api/bsp/nfc_switch
2022-11-12 18:40:59.849 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] GET api/bsp/nfc_switch performed, status: 404
2022-11-12 18:40:59.850 ERROR (MainThread) [custom_components.huawei_mesh_router.update_coordinator] Unexpected error fetching Huawei AX3 data: Expecting value: line 1 column 1 (char 0)
File "/config/custom_components/huawei_mesh_router/update_coordinator.py", line 125, in async_update
File "/config/custom_components/huawei_mesh_router/update_coordinator.py", line 149, in _update_nfc_switch
File "/config/custom_components/huawei_mesh_router/huaweiapi.py", line 303, in get
File "/config/custom_components/huawei_mesh_router/huaweiapi.py", line 81, in _get_response_json
2022-11-12 18:40:59.852 DEBUG (MainThread) [custom_components.huawei_mesh_router.update_coordinator] Finished fetching Huawei AX3 data in 0.119 seconds (success: False)
2022-11-12 18:41:19.853 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] New instance of HuaweiApi created
2022-11-12 18:41:19.854 DEBUG (MainThread) [custom_components.huawei_mesh_router.update_coordinator] Update started
2022-11-12 18:41:19.854 DEBUG (MainThread) [custom_components.huawei_mesh_router.update_coordinator] Updating router data
2022-11-12 18:41:19.854 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] Authentication started
2022-11-12 18:41:19.855 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] Refresh session called
2022-11-12 18:41:19.855 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] Session created
2022-11-12 18:41:19.855 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] Getting index
2022-11-12 18:41:19.855 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] Performing GET to html/index.html#/login
2022-11-12 18:41:19.928 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] GET html/index.html#/login performed, status: 200
2022-11-12 18:41:19.931 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] Csrf updated: param is 'lqWl7SzLaEGEN2pGF3nAl7GyLdL70Keb', token is '3rSPJpmuay30OOZ8UYUtwmvz80h8lSIA'
2022-11-12 18:41:19.932 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] Session cookies stored
2022-11-12 18:41:19.932 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] Sending nonce
2022-11-12 18:41:19.932 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] Performing POST to api/system/user_login_nonce
2022-11-12 18:41:19.940 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] POST to api/system/user_login_nonce performed, status: 200
2022-11-12 18:41:19.941 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] Csrf updated: param is 'NFMForSxE5njYjTYfdLs81CODX4uBcDm', token is '07HQuC3xA5WReuckz3lQOyU7qqGPVynk'
2022-11-12 18:41:19.943 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] Sending proof
2022-11-12 18:41:19.943 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] Performing POST to api/system/user_login_proof
2022-11-12 18:41:19.952 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] POST to api/system/user_login_proof performed, status: 200
2022-11-12 18:41:19.952 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] Csrf updated: param is '0YcpthGYVIHbrDFKRsqSEu4t0DIcO43b', token is 'eSLV75Dw6TStZZTnfMJPyr6IUhcm8EVv'
2022-11-12 18:41:19.953 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] Authentication success
2022-11-12 18:41:19.953 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] Performing GET to api/system/deviceinfo
2022-11-12 18:41:19.968 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] GET api/system/deviceinfo performed, status: 200
2022-11-12 18:41:19.970 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] Csrf updated: param is 'i3BMcu7HqtjGKqAayWOiDYGLaU0XZsPG', token is 'ps00N0QlVxlvu0Z3TQnjkgvICi00D0Au'
2022-11-12 18:41:19.971 DEBUG (MainThread) [custom_components.huawei_mesh_router.update_coordinator] Router data updated
2022-11-12 18:41:19.971 DEBUG (MainThread) [custom_components.huawei_mesh_router.update_coordinator] Updating nfc switch state
2022-11-12 18:41:19.971 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] Performing GET to api/bsp/nfc_switch
2022-11-12 18:41:19.981 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] GET api/bsp/nfc_switch performed, status: 404
2022-11-12 18:41:19.981 ERROR (MainThread) [custom_components.huawei_mesh_router.update_coordinator] Unexpected error fetching Huawei AX3 data: Expecting value: line 1 column 1 (char 0)
File "/config/custom_components/huawei_mesh_router/update_coordinator.py", line 125, in async_update
File "/config/custom_components/huawei_mesh_router/update_coordinator.py", line 149, in _update_nfc_switch
File "/config/custom_components/huawei_mesh_router/huaweiapi.py", line 303, in get
File "/config/custom_components/huawei_mesh_router/huaweiapi.py", line 81, in _get_response_json
2022-11-12 18:41:19.986 DEBUG (MainThread) [custom_components.huawei_mesh_router.update_coordinator] Finished fetching Huawei AX3 data in 0.132 seconds (success: False)
2022-11-12 18:41:59.989 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] New instance of HuaweiApi created
2022-11-12 18:41:59.989 DEBUG (MainThread) [custom_components.huawei_mesh_router.update_coordinator] Update started
2022-11-12 18:41:59.989 DEBUG (MainThread) [custom_components.huawei_mesh_router.update_coordinator] Updating router data
2022-11-12 18:41:59.989 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] Authentication started
2022-11-12 18:41:59.989 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] Refresh session called
2022-11-12 18:41:59.990 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] Session created
2022-11-12 18:41:59.990 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] Getting index
2022-11-12 18:41:59.990 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] Performing GET to html/index.html#/login
2022-11-12 18:42:00.060 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] GET html/index.html#/login performed, status: 200
2022-11-12 18:42:00.062 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] Csrf updated: param is 'EfQydw20Bsa5INo0giqI41YG2lsXwr9y', token is 'TnOoXPnex1RTsfmq7jyqFnYUzH8bFCoW'
2022-11-12 18:42:00.063 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] Session cookies stored
2022-11-12 18:42:00.063 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] Sending nonce
2022-11-12 18:42:00.063 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] Performing POST to api/system/user_login_nonce
2022-11-12 18:42:00.071 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] POST to api/system/user_login_nonce performed, status: 200
2022-11-12 18:42:00.071 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] Csrf updated: param is 'KQMnX2aU8ho38SvTA8nugDI6waShortp', token is 'Nf72xsjIjHa4t2dMwOMmPNWZ4yDLsYW8'
2022-11-12 18:42:00.074 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] Sending proof
2022-11-12 18:42:00.074 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] Performing POST to api/system/user_login_proof
2022-11-12 18:42:00.083 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] POST to api/system/user_login_proof performed, status: 200
2022-11-12 18:42:00.085 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] Csrf updated: param is 'MZoJbf0a0k1x6Ko6y7IvrOxxHO0Ns0A0', token is 'yD8TyKYI3Pj9wFhSlN040GGZlZ75PFV0'
2022-11-12 18:42:00.085 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] Authentication success
2022-11-12 18:42:00.085 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] Performing GET to api/system/deviceinfo
2022-11-12 18:42:00.098 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] GET api/system/deviceinfo performed, status: 200
2022-11-12 18:42:00.101 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] Csrf updated: param is '4DafHSFPqXSrJQZsZjPKQDonmMe3YnPk', token is 'q2q0mBZhCg1P0cuPdiMz0xPUcWf4G6a2'
2022-11-12 18:42:00.101 DEBUG (MainThread) [custom_components.huawei_mesh_router.update_coordinator] Router data updated
2022-11-12 18:42:00.101 DEBUG (MainThread) [custom_components.huawei_mesh_router.update_coordinator] Updating nfc switch state
2022-11-12 18:42:00.101 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] Performing GET to api/bsp/nfc_switch
2022-11-12 18:42:00.106 DEBUG (MainThread) [custom_components.huawei_mesh_router.huaweiapi] GET api/bsp/nfc_switch performed, status: 404
2022-11-12 18:42:00.106 ERROR (MainThread) [custom_components.huawei_mesh_router.update_coordinator] Unexpected error fetching Huawei AX3 data: Expecting value: line 1 column 1 (char 0)
File "/config/custom_components/huawei_mesh_router/update_coordinator.py", line 125, in async_update
File "/config/custom_components/huawei_mesh_router/update_coordinator.py", line 149, in _update_nfc_switch
File "/config/custom_components/huawei_mesh_router/huaweiapi.py", line 303, in get
File "/config/custom_components/huawei_mesh_router/huaweiapi.py", line 81, in _get_response_json
2022-11-12 18:42:00.111 DEBUG (MainThread) [custom_components.huawei_mesh_router.update_coordinator] Finished fetching Huawei AX3 data in 0.122 seconds (success: False)
```

**Comment by vmakeev** [comment]
The problem is that the component failed to get the state of the NFC switch.

```
Updating nfc switch state
Performing GET to api/bsp/nfc_switch
GET api/bsp/nfc_switch performed, status: 404 // HTTP code 404 (not found)
```

Three possible reasons for this behavior come to mind:

1. Your router model does not have NFC
2. Your router's API provides information about the state of the NFC switch in some other way (not bsp/nfc_switch)
3. For some reason, the router considered the request not authorized (although authentication was successful), and for this reason returned 404

The component was developed for the WS8100-23 router (the only one I had available), but I was hoping that it could work with other models.

Can you specify the exact model of the router you are using?

**Comment by timo-mart** [comment]
My router is the [Huawei WiFi AX3](https://consumer.huawei.com/en/routers/ax3-dual-core/), the dual core model I believe.

It is not necessarily a mesh router but it seems to use the same firmware.

**Comment by timo-mart** [comment]
> My router is the [Huawei WiFi AX3](https://consumer.huawei.com/en/routers/ax3-dual-core/)

On that site at least it doesn't mention anything about NFC. Seems like only the quad core model has NFC.
Would it be possible to make the NFC switch in the integration optional?

**Comment by timo-mart** [comment]
it says this on the [page for the quad core model](https://consumer.huawei.com/en/routers/ax3-quad-core/):

> Supported by NFC-enabled Android phones. Certain Android phones with non-standard NFC may not support Huawei Share. AX3 quad-core version supports Huawei Share, yet the dual-core version does not support.

**Comment by vmakeev** [comment]
Try using the component from [this branch](https://github.com/vmakeev/huawei_mesh_router/tree/feature/temp_no_nfc). In this version, NFC is completely disabled, so we can check the compatibility of other functions with your router model.

I plan to make autodetection of available functions in the future, but this will take some time

**Comment by vmakeev** [comment]
It's a good news :)

You can continue to use this version until a new version of the component is released that supports both your model and models with NFC

**Comment by ikedinha** [comment]
Would it be possible to do some control to reboot AX3 WS7200?

**Comment by vmakeev** [comment]
You can open an issue, I'll deal with it as soon as possible

**Comment by ostap7474** [comment]
Аутентификация работает, и интеграция появляется в меню, но показывает ошибку. У меня роутер Huawei WiFi AX3. Читал выше, что с таким же роутером удалось подключиться.
![2](https://user-images.githubusercontent.com/72063931/203258298-a6bf6c81-4ebd-4082-be20-0f912c872d69.jpg)

**Comment by vmakeev** [comment]
@timo-mart the bug was fixed in [release v0.7.5](https://github.com/vmakeev/huawei_mesh_router/releases/tag/v0.7.5), both routers with and without NFC are now supported

**Comment by vmakeev** [comment]
@ostap7474 Доделал автоопределение доступных фич в [релизе v0.7.5](https://github.com/vmakeev/huawei_mesh_router/releases/tag/v0.7.5), теперь всё должно корректно работать и с поддержкой NFC, и без неё.
